protocol-tests/swagger/constant-and-from-client-parameters.json

```diff
@@ -0,0 +1,142 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Constant And Client Parameter Service",
+    "description": "Constant And Client Parameter Service",
+    "version": "1.0.0"
+  },
+  "host": "localhost:3000",
+  "schemes": [
+    "http"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "paths": {
+    "/constant/clientparam/path": {
+      "put": {
+        "operationId": "putClientConstants",
+        "description": "Pass constants from the client to this function. Will pass in constant path, query, and header parameters.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/HeaderRequiredClientConstant"
+          },
+          {
+            "$ref": "#/parameters/QueryRequiredDefaultValueClientParam"
+          },
+          {
+            "$ref": "#/parameters/QueryRequiredClientParam"
+          },
+          {
+            "$ref": "#/parameters/QueryRequiredClientConstant"
+          },
+          {
+            "$ref": "#/parameters/QueryNonRequiredClientConstant"
+          },
+          {
+            "$ref": "#/parameters/QueryRequiredConstant"
+          },
+          {
+            "$ref": "#/parameters/QueryNonRequiredConstant"
+          }
```

Comment:
separate the case of "constant" and "from client" (don't specify both in same paramter).

1. required client
2. non-required client
3. require constant (but not client)
4. non-required constant (which likely not a constant)